sqlitecluster/SQLiteClusterMessenger.h

```diff
@@ -42,6 +57,12 @@ class SQLiteClusterMessenger {
     // Checks if a command will cause the server to close this socket, indicating we can't reuse it.
     static bool commandWillCloseSocket(BedrockCommand& command);
 
+    // Establishes a connection to the host associated with the socket and actually sends the message.
+    bool _sendCommandOnSocket(SHTTPSManager::Socket& socket, BedrockCommand& command);
```

Comment:
Yeah `static` seems good 👍🏼

Reply:
Oh hm, not sure if this will work, it seems to be complaining about calling member functions and using member variables in the function. I think `const` will work, but not 100% sure yet, got some other complier errors.

Reply:
When I try `const`, I get this error and a handful of other ones:

```
sqlitecluster/SQLiteClusterMessenger.cpp: In member function ‘bool SQLiteClusterMessenger::_sendCommandOnSocket(STCPManager::Socket&, BedrockCommand&) const’:
sqlitecluster/SQLiteClusterMessenger.cpp:148:75: error: passing ‘const SQLiteClusterMessenger’ as ‘this’ argument discards qualifiers [-fpermissive]
  148 |         WaitForReadyResult result = waitForReady(fdspec, command.timeout());
      |                                                                           ^
sqlitecluster/SQLiteClusterMessenger.cpp:29:44: note:   in call to ‘SQLiteClusterMessenger::WaitForReadyResult SQLiteClusterMessenger::waitForReady(pollfd&, uint64_t)’
   29 | SQLiteClusterMessenger::WaitForReadyResult SQLiteClusterMessenger::waitForReady(pollfd& fdspec, uint64_t timeoutTimestamp) {
      |                                            ^~~~~~~~~~~~~~~~~~~~~~
sqlitecluster/SQLiteClusterMessenger.cpp:198:51: error: passing ‘const SQLiteClusterMessenger’ as ‘this’ argument discards qualifiers [-fpermissive]
  198 |         if (waitForReady(fdspec, command.timeout()) != WaitForReadyResult::OK) {
      |                                                   ^
sqlitecluster/SQLiteClusterMessenger.cpp:29:44: note:   in call to ‘SQLiteClusterMessenger::WaitForReadyResult SQLiteClusterMessenger::waitForReady(pollfd&, uint64_t)’
   29 | SQLiteClusterMessenger::WaitForReadyResult SQLiteClusterMessenger::waitForReady(pollfd& fdspec, uint64_t timeoutTimestamp) {
      |                                            ^~~~~~~~~~~~~~~~~~~~~~
```

When I try `static`, it complains about accessing member functions inside of this one. Any idea how to fix or just leave it non-const/static?

Reply:
It looks like you can't call `waitForReady` from a `const` function because `waitForReady` is not a `const` function. But I think `waitForReady` can be `static` and that alleviates this issue.